dashboards/clicks_and_impressions.py

```diff
@@ -0,0 +1,64 @@
+import pandas as pd
+import streamlit as st
+from auxiliary_functions import extract_query_params, process_query_params
+
+st.title("Clicks & Impressions Dashboard")
+
+# Initialize connection
+conn = st.connection("mysql", type="sql")
+
+df = conn.query("select * from website_event;")
+
+df_processed = df.copy(deep=True)
+#
+df_processed["url"] = "/?" + df["url_query"].astype(str)
+df_processed["query_params"] = df_processed["url"].apply(extract_query_params)
+df_processed["parsed_params"] = df_processed["query_params"].apply(process_query_params)
+
+# helper to extract industry
+def extract_industry(params):
+    industries = params.get("industries", [])
+    if isinstance(industries, list) and industries:
+        return industries[0]
+    return None
+
+df_processed["industries"] = df_processed["parsed_params"].apply(extract_industry)
```

Comment:
From what I understand of this code, a click/impression on a mentor is considered to be of Industry X if the search query parameter includes Industry X. However, this isn't necessarily the case - in fact, the URL visit isn't sufficient and one needs to use the Elasticsearch data on individual mentors in order to obtain their industry (see https://github.com/AdvisorySG/mentorship-streamlit/issues/42).